network/src/network.rs

```diff
@@ -698,7 +703,11 @@ pub struct NetworkService {
 }
 
 impl NetworkService {
-    pub fn new(network_state: Arc<NetworkState>, protocols: Vec<CKBProtocol>) -> NetworkService {
+    pub fn new(
+        network_state: Arc<NetworkState>,
+        protocols: Vec<CKBProtocol>,
+        consensus: &Consensus,
+    ) -> NetworkService {
```

Comment:
IdentifyCallback only cares the service name, so passing `name: String` instead of `consensus: Consensus`, is enough.

Reply:
And can we try to reuse [`code_name`](https://github.com/nervosnetwork/ckb/blob/a5ce3dfe881dddf8a798db7f707ae0195c6e4ada/src/main.rs#L38) as service name here?

Reply:
Can't do this, now the name is `/chain_id/genesis_hash[..8]`.

`code_name` change does not represent `chan_id` and `genesis_hash` change

Reply:
> IdentifyCallback only cares the service name, so passing `name: String` instead of `consensus: Consensus`, is enough.

I am not sure if the data in genesis is still needed in the future. Passing the entire reference has no direct impact on the performance of the code.

Reply:
> > IdentifyCallback only cares the service name, so passing `name: String` instead of `consensus: Consensus`, is enough.
> 
> I am not sure if the data in genesis is still needed in the future. Passing the entire reference has no direct impact on the performance of the code.

Not about the performance.
It avoids introducing `ckb-chain-spec` as a dependency if only use the name as the parameter.